src/components/Table/Table.ColumnChooser.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import equal from 'fast-deep-equal'
+import { noop } from '../../utilities/other'
+import DropList from '../DropList'
+import { IconBtn } from '../DropList/DropList.togglers'
+import { columnShape, createColumnChooserListItems } from './Table.utils'
+
+const shouldColumnShow = column => !!column.show
+
+function ColumnChooser({
+  columns,
+  defaultColumns,
+  columnChooserResetLabel = 'Reset to defaults',
+  onColumnChoose = noop,
+  resetColumns = noop,
+  updateColumns = noop,
+}) {
+  const items = createColumnChooserListItems(columns)
+
+  return (
+    <DropList
+      autoSetComboboxAt={10}
+      items={items}
+      onSelect={(selection, clickedItem) => {
+        if (clickedItem.type === 'reset_droplist') {
+          if (!equal(columns, defaultColumns)) {
+            resetColumns(defaultColumns)
+          }
+          onColumnChoose(clickedItem, defaultColumns.filter(shouldColumnShow))
+        } else {
+          updateColumns(columns, clickedItem)
+          onColumnChoose(clickedItem, selection)
+        }
+      }}
+      selection={columns.filter(shouldColumnShow)}
+      toggler={
+        <IconBtn
+          a11yLabel="Choose columns to show or hide"
+          iconName="column-check"
+          iconSize={24}
+          withCaret={false}
+        />
+      }
+      withMultipleSelection
+      withResetSelectionItem={columnChooserResetLabel}
```

Comment:
I agree... I mention the reasons in the previous comment (and accept suggestions). I just really dislike adding a gazillion amount of props.

Reply:
It can definitely be left as is, there's a reason we have PropTypes and people will figure it out like I did. The only alternative I could think of is leaving off the `with`, so it's just `resetSelectionItem`.